### PR TITLE
Fix Add Patient Bugs

### LINKED ISSUE #98: FIx add patient bugs
Delimiter to use backslash

Override hashcode for those place where equals is overriden

### PR DESCRIPTION
**Pull request by cheeggered**
Closes #98

### CODE REVIEW

File: src/main/java/seedu/address/model/person/NameContainsKeywordsPatientPredicate.java [hunk 1]
```diff
@@ -16,6 +17,10 @@ public NameContainsKeywordsPatientPredicate(List<String> keywords) {
         this.keywords = keywords;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.keywords);
+    }
     @Override
```
Comment (yanghengtang) [inline src/main/java/seedu/address/model/person/NameContainsKeywordsPatientPredicate.java:24]: Extra empty line before this override statement

**Review by yanghengtang — COMMENTED** [review]
Delimiter for parameter not changed yet

File: src/test/java/seedu/address/logic/commands/AddPatientCommandTest.java [hunk 2]
```diff
@@ -29,6 +32,18 @@
...
 
+    @Test
+    public void hashcode() {
+        AddPatientCommand command = new AddPatientCommand(ALICE);
+
+        // same value -> returns same hashcode
+        assertEquals(command.hashCode(), Objects.hash(ALICE));
+
+        // different value -> returns different hashcode
+        assertNotEquals(command.hashCode(), Objects.hash(BOB));
```
Comment (derekjxtan) [inline src/test/java/seedu/address/logic/commands/AddPatientCommandTest.java:43]: remove this test as this is not expected behaviour. java documentation states: `If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result.`

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPatientPredicateTest.java [hunk 3]
```diff
@@ -2,18 +2,34 @@
...
+        List<String> keywords = List.of("keyword1", "keyword2");
+
+        // same value -> return same hashcode
+        NameContainsKeywordsPatientPredicate command = new NameContainsKeywordsPatientPredicate(keywords);
+        assertEquals(command.hashCode(), Objects.hash(keywords));
+
+        List<String> keywords2 = List.of("keyword1", "keyword3");
+
+        // different value -> returns different hashcode
+        assertNotEquals(command.hashCode(), Objects.hash(keywords2));
```
Comment (derekjxtan) [inline src/test/java/seedu/address/model/person/NameContainsKeywordsPatientPredicateTest.java:30]: likewise

**Review by derekjxtan — COMMENTED** [review]
add tests for AddPatientCommandParser as well

**Review by derekjxtan — APPROVED** [review]
lgtm

**Review by derekjxtan — APPROVED** [review]
lgtm

**Pull request merged**